persistence/src/main/java/org/mskcc/smile/persistence/neo4j/SmileSampleRepository.java

```diff
@@ -89,4 +89,12 @@ List<SampleMetadata> findSampleMetadataHistoryByNamespaceValue(
             + "CREATE (s)<-[:HAS_SAMPLE]-(p)")
     void updateSamplePatientRelationship(@Param("smileSampleId") UUID smileSampleId,
             @Param("smilePatientId") UUID smilePatientId);
+
+    @Query("MATCH (s: Sample {sampleCategory: 'research'})-[:HAS_METADATA]->(sm: SampleMetadata) "
+            + "WHERE sm.importDate > $inputDate RETURN DISTINCT s.smileSampleId")
+    List<UUID> findSamplesByLatestImportDate(@Param("inputDate") String inputDate);
```

Comment:
Curious, does Cipher provide a way to return the metadata here without a subsequent fetch below?

Reply:
with cypher directly - yes but I wasn't able to with the spring-data client unfortunately and wasn't able to find any documentation supporting that specific use case